src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/peripheral/RSBridgePeripheral.java

```diff
@@ -0,0 +1,893 @@
+            if (parsedFilter.testRS(new ResourceAmount(status.info().resource(), 1))) {
+                craftingManager.cancel(status.info().id());
+                canceled++;
+            }
+        }
+
+        return MethodResult.of(canceled);
+    }
+
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult isCraftable(IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        Pair<? extends GenericFilter<?>, String> filter = GenericFilter.parseGeneric(arguments.getTable(0));
+        if (filter.getRight() != null)
+            return MethodResult.of(null, filter.getRight());
+
+        GenericFilter<?> parsedFilter = filter.getLeft();
+
+        return MethodResult.of(RsApi.findPatternFromFilters(getNetwork(), null, parsedFilter).getLeft() != null);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult isCrafting(IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        Pair<? extends GenericFilter<?>, String> filter = GenericFilter.parseGeneric(arguments.getTable(0));
+        if (filter.getRight() != null)
+            return MethodResult.of(null, filter.getRight());
+
+        GenericFilter<?> parsedFilter = filter.getLeft();
+
+        AutocraftingNetworkComponent craftingManager = getComponent(AutocraftingNetworkComponent.class);
+
+        for (TaskStatus status : craftingManager.getStatuses()) {
+            if (parsedFilter.testRS(new ResourceAmount(status.info().resource(), 1))) {
+                return MethodResult.of(true);
+            }
+        }
+
+        return MethodResult.of(false);
+    }
+
+    @Override
+    @LuaFunction(mainThread = true)
+    public final MethodResult getPatterns(IArguments arguments) throws LuaException {
+        if (!isAvailable())
+            return notConnected();
+
+        // Expected input is a table with either an input table, an output table or both to filter for both
+        // If no table is provided or it's empty, return every pattern
+        Map<?, ?> filterTable = arguments.optTable(0, Collections.emptyMap());
+        if (filterTable.isEmpty()) {
+            return MethodResult.of(RsApi.getPatterns(getNetwork()));
+        }
+
+        boolean hasInputFilter = filterTable.containsKey("input");
+        boolean hasOutputFilter = filterTable.containsKey("output");
+        boolean hasAnyFilter = hasInputFilter || hasOutputFilter;
+
+        // If the player tries to filter for nothing, return nothing.
+        if (!hasAnyFilter)
+            return MethodResult.of(null, "NO_FILTER");
+
+        GenericFilter<?> inputFilter = null;
+        GenericFilter<?> outputFilter = null;
+
+        if (hasInputFilter) {
+            Map<?, ?> inputFilterTable = TableHelper.getTableField(filterTable, "input");
+
+            inputFilter = GenericFilter.parseGeneric(inputFilterTable).getLeft();
```

Comment:
did you check if the filter is valid or is it being ignored on purpose

Reply:
I didn't really think about it. It would return an empty filter by default so the function would not break
But I guess returning the debug message would be good